nsv13/code/modules/munitions/ship_weapons/energy_weapons/laser_pd.dm

```diff
@@ -0,0 +1,98 @@
+// Laser PD system control console
+/obj/machinery/computer/laser_pd
+	name = "point-defense laser console"
+	desc = "A computer that allows you to control a connected point-defense laser."
+	icon_screen = "50cal"
+	circuit = /obj/item/circuitboard/computer/laser_pd
+	var/obj/machinery/ship_weapon/energy/laser_pd/turret
+	var/gun_id = 0 // Used for map linkage
+
+/obj/machinery/computer/laser_pd/Initialize(mapload)
+	..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/computer/laser_pd/LateInitialize()
+	. = ..()
+	for(var/obj/machinery/ship_weapon/energy/laser_pd/pd_gun in GLOB.machines)
+		if(gun_id && pd_gun.gun_id == gun_id)
+			turret = pd_gun
+			break
+
+/obj/machinery/computer/laser_pd/attack_robot(mob/user)
+	. = ..()
+	return attack_hand(user)
+
+/obj/machinery/computer/laser_pd/attack_ai(mob/user)
+	. = ..()
+	return attack_hand(user)
+
+/obj/machinery/computer/laser_pd/multitool_act(mob/living/user, obj/item/multitool/tool)
+	..()
+	. = TRUE
+	var/obj/machinery/ship_weapon/energy/laser_pd/stored_gun = tool.buffer
+	if(stored_gun && istype(stored_gun))
+		turret = stored_gun
+		to_chat(user, "<span class='warning'>Successfully linked [src] to [turret].")
+	else
+		to_chat(user, "<span class='warning'>No compatible weapon type found in buffer.")
+
+/obj/machinery/computer/laser_pd/attack_hand(mob/user)
+	. = ..()
+	if(machine_stat & (BROKEN|NOPOWER))
+		to_chat(user, "<span class='warning'>[src] doesn't seem to be working.</span>")
+		return
+	if(!turret)
+		to_chat(user, "<span class='warning'>[src] is not linked to a laser turret! You can link it with a multitool.</span>")
+		return
+	turret.start_gunning(user)
+
+// Overmap gunning component for laser PD
+/datum/component/overmap_gunning/laser_pd
+	fire_mode = FIRE_MODE_LASER_PD
+	fire_delay = 1.5 SECONDS
+
+// The laser PD turret itself
+/obj/machinery/ship_weapon/energy/laser_pd
+	name = "laser point defense turret"
+	desc = "A low-wattage laser designed to provide close-combat relief to large ships."
+	icon ='nsv13/icons/obj/energy_weapons.dmi'
+	icon_state = "blaster_cannon"
```

Comment:
Why's the icon name unrelated to the actual thing name?

Reply:
That's the name of the original file I was given for the cannon, but I can change it.